kiwi/debug.h

```diff
@@ -54,64 +54,51 @@ class DebugHelper
 
     static void dump(const SolverImpl::RowMap &rows, std::ostream &out)
     {
-        typedef SolverImpl::RowMap::const_iterator iter_t;
-        iter_t end = rows.end();
-        for (iter_t it = rows.begin(); it != end; ++it)
+        for (const auto& rowPair : rows)
```

Comment:
```suggestion
        for (const auto &rowPair : rows)
```

Following the convention of the rest of the program, make the reference (and pointer, if they occur) tokens hug the identifier rather than the type.

This is (usually) because a declaration list requires such tokens for each identifier, e.g.

```c++
int& foo, bar;
```

is really

```c++
int &foo;
int bar;
```

instead of 

```c++
int &foo;
int &bar;
```

Thus hugging the identifier with the token instead of the type helps to convey this.

Just a nitpick.

Reply:
You're right, I should follow the rest of the program conventions, and I will update on this.
Although I think `*` and `&` don't belong to the var name, for the very simple reason they're not the same type.

`T*` and `T` are both very different type with very different semantics, same goes for `T&` and `T` (and it's even worse with `T&&`).

As for variables declaration, I think this is just inherited from C, and is mostly broken. I never declare two variables on the same line because of that.